fusesoc/edalizer.py

```diff
@@ -240,6 +244,12 @@ def create_edam(self):
         for snippet in first_snippets + snippets + last_snippets:
             merge_dict(self.edam, snippet)
 
+    def clean_temp_dirs(self):
+        for core in self.cores:
+            if core.core_root.startswith(tempfile.gettempdir()):
+                logger.debug(f"Removing {core.core_root} ttptttg temporary directory")
```

Comment:
Very clever!I'm just wondering if there could be any cases where the cores are stored in the temp dir and we don't want to remove them. Perhaps I'm being a bit paranoid here

Reply:
We could keep track of the cores created by generators, and selectively remove only their `core_root`

Reply:
That would be great if you think we can do it that way